x-pack/metricbeat/module/mssql/connection.go

```diff
@@ -0,0 +1,26 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package mssql
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func NewConnection(uri string) (*sql.DB, error) {
+	db, err := sql.Open("sqlserver", uri)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create db instance")
+	}
+
+	// Check the connection before executing all queries to reduce the number
+	// of connection errors that we might encounter.
+	if err = db.Ping(); err != nil {
+		err = errors.Wrap(err, "error doing ping to db")
+	}
+
+	return db, err
```

Comment:
This db instance is closed at the Metricset, I'm not fully sure if this is correct.

Reply:
Is the connection per metricset? If yes, then this sounds correct.

Reply:
Right now it is per Metricset yes. I wasn't sure if that was correct either :sweat_smile: but it looked like the best way to isolate errors between metricset (that each has it's own connection).